src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js

```diff
@@ -50,7 +52,13 @@ const customHTMLElementModels = {
     }),
 };
 
-const defaultViewProps = {style: [styles.alignItemsStart, styles.userSelectText]};
+// For web platform defaultViewProps should use block display, otherwise immediate
+// children will inherit display:block even when they have display:inline set in CSS.
+const defaultViewProps = {
```

Comment:
`defaultViewProps` was using flex display here. This forces all immediate children to be block displays by default. This was the reason all comment blocks to behave like blocks, even when the had `display:inline` style given as an inline style. So hyperlinks were rendered as block elements and `cursor: pointer` style was applied to the block element. This caused pointer cursor to be shown even in empty areas.

Display block can be applied only for web & desktop platforms. That's the reason we are switching the styles using a ternary operator here.